code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -208,6 +209,15 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 	/// Cached value of is_station_intact.
 	var/cached_station_intact = TRUE
 
+	/// If not null, use this instead of world.time
+	var/simulated_time = null
```

Comment:
just add a proc to get the dynamic time. this is really messy

Reply:
I did this specifically to avoid the proc overhead but I can do anyway since it shouldn't be super hot